src/components/AddItem.js

```diff
@@ -1,11 +1,102 @@
-import React from 'react';
+import React, { useState } from 'react';
+import { useHistory } from 'react-router-dom';
+import { db } from '../lib/firebase';
+
+const initialFormState = {
+  itemName: '',
+  timeFrame: 7,
+  lastPurchased: null,
+};
 
 const AddItem = () => {
+  const token = 'user token';
+  const history = useHistory();
+  const [formData, setFormData] = useState(initialFormState);
+  const [error, setError] = useState('');
+  console.log(error);
+
+  const handleFormChange = (event) => {
+    event.preventDefault();
+    setFormData({
+      ...formData,
+      [event.target.name]: event.target.value,
+    });
+  };
+
+  const handleSubmit = (event) => {
+    event.preventDefault();
+    if (!formData.itemName) {
+      setError(`${event.target.name} field cannot be empty`);
+    } else {
+      db.collection(token).add(formData);
+      history.push('/list');
+      setFormData(initialFormState);
```

Comment:
Hey @DeeDowns @rachelrohrbach I have a hunch that this might be the source of your error:
```
Warning: Can't perform a React state update on an unmounted component.
```
Notice how it says we are performing a state update (e.g. setFormData) _after_ the component has already been unmounted. This matches with the code I'm seeing here. Because this component unmounts when we change route and go to the /list page, when we try to update the state right afterwards we might end up trying to update state on an unmounted component. 
Maybe you could try moving this line of code above the `history.push` call?

Reply:
I can't verify that this actually fixes the warning, so I'm not totally sure. Still, it might be worth doing anyways, just in case. I'd put it down as the last line of this function, so that setting formData and Error are both done before navigation away from this page.

Reply:
> I can't verify that this actually fixes the warning, so I'm not totally sure. Still, it might be worth doing anyways, just in case. I'd put it down as the last line of this function, so that setting formData and Error are both done before navigation away from this page.

I moved them both about history.push but still get the warning